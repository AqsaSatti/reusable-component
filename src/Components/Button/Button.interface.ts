export interface ButtonProps{
    children?:string;
    variant?:string;
    startIcon?:string;
    // endIcon?:React.ReactNode;
    endIcon?:string;
    href?:string;
    disabled?:boolean;
    onClick?:  (event: React.MouseEvent<HTMLButtonElement>) => void; 
    color?:string;
    size?:string;
}