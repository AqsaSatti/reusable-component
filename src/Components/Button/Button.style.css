.contained{
    background-color: blue;
    color: white;
    border: none;
    /* display: inline; */
    width: 150px;
    margin-bottom: 20px;
    border-radius: 5px;
    text-align: center;
    
    /* cursor: pointer; */
  }
  .contained:hover{
    background-color: rgb(67, 28, 123);
  }
  
  button:disabled{
    background-color: rgb(192, 192, 226);
    width: 150px;
    margin-bottom: 20px;
    border: none;
    /* opacity: 0.5; */
    /* cursor: not-allowed; */
    border-radius: 5px;
  }
  .link{
    /* display: inline; */
    width: 100px;
    margin-bottom: 20px;
    background-color: blue;
    color: white;
    border: none;
    border-radius: 5px;
    padding: 4px;
    cursor: pointer;
  
  }
  .outlined{
    border: 2px solid blue;
    width: 150px;
    margin-bottom: 20px;
    border-radius: 5px;
  }
  .small {
    padding: 7px 12px;
    font-size: 17px;
    border-radius: 5px;
  }
  
  .medium {
    padding: 8px 16px;
    font-size: 17px;
  }
  
  .large {
    padding: 12px 24px;
    font-size: 16px;
  
  }
  .delete{
    background-color: rgb(122, 164, 233);
    color: white;
    border: none;
    width: 150px;
    margin-bottom: 20px;
    border-radius: 5px;
    text-align: center;
  }
  .send{
    background-color: blue;
    color: white;
    border: none;
    width: 150px;
    border-radius: 9px;
    text-align: center;
    margin-bottom: 20px;
  }
  .red{
  color: red;
  background-color: white;
  border: 2px solid red;
  }
  .green{
  background-color: green;
  color: white;
  
  }
  .start_icon_button{
    margin-right: 8px; 
  }
  
  .end_icon_button {
    margin-left: 8px;
  }
  
  /* Optional: Adjust the size of the icon if needed */
  /* button i {
    font-size: 1.2em; 
  } */