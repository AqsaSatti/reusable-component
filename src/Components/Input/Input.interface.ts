
import { ChangeEvent, HTMLInputTypeAttribute } from 'react';

export interface InputProps {
  variant?: 'outlined' | 'filled' | 'underlined' | 'rounded' ; 
  placeholder?: string; 
  label?: string; 
  value?: string;
  name?: string;
  autoComplete?: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  type?: HTMLInputTypeAttribute; // Support for different input types (text, password, etc.)
  // type?: 'text' | 'password' | 'number';
  disabled?: boolean;
  required?: boolean; // For required input fields
  defaultValue?: string;
  // pattern?: string; // For regex pattern validation
  fullWidth?: boolean; // Optional prop to make the input full-width
  error?: boolean;
  errorMessage?: string;
  startIcon?:string;
  endIcon?:string;
  className?:string;
}