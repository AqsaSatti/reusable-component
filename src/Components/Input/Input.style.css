:root {
  --red: red;
  --blue: #007bff;
  --bg-color: #f1f1f1;
  --bx-shadow: rgba(0, 123, 255, 0.5);
  --border: 2px solid var(--blue);
  --icon-color: blueviolet;
  --gap: 8px;
  --rounded-border-radius: 25px;
  --icon-spacing: 35px;
  --input-width: 250px;
  --input-height: 45px;
  --icon-position-top: 25%;
}

.input-container {
  /* position: relative; */
  display: flex;
  flex-direction: column;
  gap: var(--gap);
}

.input-label {
  font-size:16px;
  color: #333;
}
.input-label-error {
  color: var(--red); 
}
/* .placeholder{
  margin-left: 200px;
} */
.input {
  padding: 8px;
  font-size: 16px; 
  border-radius: 4px;
  border: 1px solid #ccc;
  width: var(--input-width); 
  height: var(--input-height);
  box-sizing: border-box; /* Ensures padding is included in the width/height */
  padding-right: 50px;
}
input.outlined {
  border: 1px solid purple;
  width: auto;
  /* border: none !important; */
}

.filled {
  background-color: var(--bg-color);
  border: none;
}

.underlined {
  border: none;
  border-bottom: var(--border);
}
.input.rounded{
border-radius: var(--rounded-border-radius) !important ;
width: auto; 
height: auto;
padding-left: 17px;
}

.input:required {
  border: 2px solid var(--red); ; 
}
.required-asterisk {
  color: var(--red); ; 
  margin-left: 4px;
}
.input:focus {
  /* border-color: inherit; Prevent border color from changing on focus */
  border-color: var(--blue); /*Blue border color on focus*/
   box-shadow: 0 0 5px var(--bx-shadow); 
   outline: none;
}
.input.full-width {
  width: 100%; /* Make the input full width */
}
.input.input-error {
  border-color: var(--red); ; 
  border: 2px solid var(--red); ;
}
.error-text {
  color: var(--red); 
  font-size: 16px;
  margin-top: -25px;

}
.start-icon {
  /* margin-right: 8px; */
  position: absolute;
  top: var(--icon-position-top);
  left: 10px; 
  color: var(--icon-color)
}
.end-icon {
  position: absolute;
  top: var(--icon-position-top);
  left: 200px; 
  margin-left: 16px;
  color:var(--icon-color);
  /* color: #aaa; change icon color  */
}

.icon-spacing {
  padding-left:var(--icon-spacing)
}
/* .input-container input {
  padding-left: 35px;
} */
.input-wrapper{
  position: relative;
}
.disabled-icon{
color:#ccc;
}
