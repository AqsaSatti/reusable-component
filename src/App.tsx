import logo from "./logo.svg";
import React from "react";
import Button from "./Components/Button/Button";
import Icon from "./Components/Icon";
import "./App.css";
 

// function App() {
//   return (
//     <div className="App">
//       {/* <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <p>
//           Edit <code>src/App.js</code> and save to reload.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//         >
//           Learn React
//         </a>
//       </header> */}
//       <h1>Reuseable Component</h1>
//       {/* <Button style={{backgroundColor: 'green', color: 'red'}} onClick={()=>{alert('button1 is clicked')}}>click me 1</Button> */}
//       <br></br>
//       {/* <Button onClick={()=>{alert('button2 is clicked')}}><h1>click me 2</h1></Button> */}
//       {/* <TextInput></TextInput> */}
//       <button></button>
//     </div>
//   );
// }

// function App() {
//   return (
//     <div className="App">
//       <h1>Reusable Button Component</h1>

//       {/* Primary Variant */}
//       <Button variant="primary" onClick={() => alert('Primary button clicked')}>
//         Primary
//       </Button>

//       {/* Secondary Variant */}
//       <Button variant="secondary" onClick={() => alert('Secondary button clicked')}>
//         Secondary
//       </Button>

//       {/* Danger Variant */}
//       <Button variant="danger" onClick={() => alert('Danger button clicked')}>
//         Danger
//       </Button>

//       {/* Disabled Button */}
//       <Button variant="primary" disabled>
//         Disabled
//       </Button>
//     </div>
//   );
// }

function App() {
  return (
    <div className="container">
      <h1 className="mt-5">Variants</h1>

      {/* <Button variant="primary" onClick={() => alert("Primary button clicked")}>
        Primary
      </Button>
      <Button variant="secondary" onClick={() => alert("Secondary button clicked")}>
        Secondary
      </Button>
      <Button variant="success" onClick={() => alert("Success button clicked")}>
        Success
      </Button> */}
      {/* /////////////////////////////////////////// */}
      <br></br>
      <div className="d-flex flex-column ">
        <Button variant="contained" size="small" onClick={() => alert("contained button clicked")}> Contained </Button>

        <Button  disabled size="medium" onClick={() => alert("disabled button clicked")}> Disabled </Button>

        <Button variant="link" href="#" size="small" > Link </Button>
        <Button variant="outlined" size="large" onClick={() => alert("outlined button clicked")}> Outlined </Button>

        {/* <Button startIcon={<i className="fa fa-trash "></i>} size="medium" onClick={() => alert("delete button clicked")}> Delete </Button> */}

        <Button startIcon={"fa-trash"}> Delete </Button>

        <Button variant="send" endIcon={"fa-paper-plane"}> Send </Button>

        <Button color = "red" > Error </Button>
        <Button color = "green" > Success </Button>
        {/* <Icon iconName = "fa-trash"></Icon> */}
        
      </div>
    </div>
  );
}

export default App;
