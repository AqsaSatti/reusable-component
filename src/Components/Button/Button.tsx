import React from 'react';
// import styles from './Button.style.module.css'
import  './Button.style.css'
import { ButtonProps } from './Button.interface';

// export interface ButtonProps {
//   label: string;
//   variant?: 'contained' | 'outlined' | 'link' | 'delete' | 'send'; // Variants defined as specific options
//   size?: 'small' | 'medium' | 'large';
//   disabled?: boolean;
//   onClick?: () => void;
// }

const Button = ({ variant = "contained", onClick, children, size, startIcon, endIcon, disabled = false, color = '' }: ButtonProps) => {

  return (
    <button
      data-testId = "btn-id"
      className={`${variant} ${size} ${color}`}
      onClick={onClick}
      disabled={disabled}
    >
      {/* {variant === "delete" && <i class="fa fa-trash " ></i>}
      {variant === "send" && <i class="fa fa-paper-plane icon-spacing" ></i>} */}

      {startIcon && (
        <span className="start_icon_button">
          {startIcon && <i className={`fa ${startIcon}`} />}
        </span>
      )}
      {children}
      {endIcon && (
        <span className="end_icon_button">
          <i className={`fa ${endIcon}`} />
        </span>
      )}
    </button>
  );
}

export default Button;
