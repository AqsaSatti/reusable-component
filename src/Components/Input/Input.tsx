import React from 'react';
import { InputProps } from './Input.interface';
import './Input.style.css';

const Input: React.FC<InputProps> = ({
  variant = 'outlined',
  placeholder = '',
  label,
  value,
  name,
  autoComplete,
  onChange,
  type = 'text',
  disabled = false,
  required = false,
  defaultValue,
  error = false,
  errorMessage,
  startIcon,
  endIcon,
  className = ''
}) => {
  // Combine base class with variant class
  const inputClass = `input ${variant} ${error ? 'input-error' : ''} ${className && className} ${startIcon ? 'icon-spacing' : ''} `;
  const labelClass = `input-label ${error ? 'input-label-error' : ''}`;
  const iconClass = `start_icon ${disabled ? 'disabled-icon' : ''} `;
  return (
    <div className="input-container">
      <label className={labelClass} >
        {label}
        {required && <span className="required-asterisk">*</span>}
      </label>

      <div className='input-wrapper'>
      
          {startIcon && (
            <span className={iconClass}>
              {startIcon && <i className={`fa ${startIcon}`} />}
            </span>
          )}
          <input
            className={`${inputClass} `}
            type={type}
            placeholder={placeholder} // Remove placeholder text if required
            value={value}
            name={name}
            onChange={onChange}
            disabled={disabled}
            required={required}
            defaultValue={defaultValue}
            autoComplete = {autoComplete}
          />
          {endIcon && (
            <span className="end_icon">
              {endIcon && <i className={`fa ${endIcon}`} />}
            </span>
          )}
        {error && <p className='error-text'>{errorMessage}</p>}
      </div>
    </div>
  );
};

export default Input;
