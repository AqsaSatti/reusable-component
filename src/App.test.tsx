import React from 'react';
import { render, screen, act, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import App from './App';
import Sum from './Sum';
import Input from './Components/Input/Input'
import Button from './Components/Button/Button'

test('renders learn react link', () => {
  render(<App />);
  const text = screen.getByText(/Variants/i);
});

test('button click', () => {
  render(<Button />);
  const button = screen.getByTestId('btn-id');

  fireEvent.click(button)
  expect(onclick).toHaveBeenCalledTimes(1)
});

// *********Sum Function TEsting*************
// test("testing for sum function", ()=>{
//   expect(Sum(10,2)).toBe(12)
// })

test("testing input", () => {

  render(<Input />);
  let checkInput = screen.getByRole("textbox") as HTMLInputElement
  // let checkInputPlaceHolder  = screen.getByPlaceholderText("nnn")
  expect(checkInput).toBeInTheDocument();
  // expect(checkInput).toHaveAttribute("nawme");
  fireEvent.change(checkInput, { target: { value: "abc" } })
  expect(checkInput.value).toBe("abc");
})

// test("click event test case", () => {
//   render(<Button />);
//   const btn = screen.getByRole("button")
//   fireEvent.click(btn)
//   expect(screen.getByText("text")).toBeInTheDocument();
// })

// ********************Describe Function************
describe.only("UI test case group1", ()=>{
  test("testting....", ()=>{
    render(<Input/>);
    let checkInput  = screen.getByRole("textbox")
    expect(checkInput).toBeInTheDocument();
  })
  test("testing for sum function", ()=>{
    expect(Sum(10,2)).toBe(12)
  })
})

describe.skip("UI test case group2", ()=>{
  test("testing for sum function", ()=>{
    expect(Sum(10,2)).toBe(12)
  })
})

//*****************Making SnapShots***************
//  test("snapshot for app component", ()=>{
//   const container = render(<App/>)
//     expect(container).toMatchSnapshot()
//   })