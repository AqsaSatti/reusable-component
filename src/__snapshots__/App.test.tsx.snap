// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for app component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="mt-5"
        >
          Variants
        </h1>
        <br />
        <div
          class="d-flex flex-column "
        >
          <button
            class="contained medium "
          >
             Contained 
          </button>
          <button
            class="contained medium "
            disabled=""
          >
             Disabled 
          </button>
          <button
            class="link small "
          >
             Link 
          </button>
          <button
            class="outlined large "
          >
             Outlined 
          </button>
          <button
            class="contained undefined "
          >
            <span
              class="start_icon_button"
            >
              <i
                class="fa fa-trash"
              />
            </span>
             Delete 
          </button>
          <button
            class="send undefined "
          >
             Send 
            <span
              class="end_icon_button"
            >
              <i
                class="fa fa-paper-plane"
              />
            </span>
          </button>
          <button
            class="contained undefined red"
          >
             Error 
          </button>
          <button
            class="contained undefined green"
          >
             Success 
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="mt-5"
      >
        Variants
      </h1>
      <br />
      <div
        class="d-flex flex-column "
      >
        <button
          class="contained medium "
        >
           Contained 
        </button>
        <button
          class="contained medium "
          disabled=""
        >
           Disabled 
        </button>
        <button
          class="link small "
        >
           Link 
        </button>
        <button
          class="outlined large "
        >
           Outlined 
        </button>
        <button
          class="contained undefined "
        >
          <span
            class="start_icon_button"
          >
            <i
              class="fa fa-trash"
            />
          </span>
           Delete 
        </button>
        <button
          class="send undefined "
        >
           Send 
          <span
            class="end_icon_button"
          >
            <i
              class="fa fa-paper-plane"
            />
          </span>
        </button>
        <button
          class="contained undefined red"
        >
           Error 
        </button>
        <button
          class="contained undefined green"
        >
           Success 
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
